### PR TITLE
Add "components" doc.

### PR DESCRIPTION
**Pull request by mtias**
Work in progress: document the tools and components we have for creating blocks.

### CODE REVIEW

File: docs/block-components.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
+# Tools
+
+To simplify block customization and ensure a consistent experience for users, there are a number of built-in UI patterns to help generate the editor preview. These common tools are offered under the `wp.blocks` global. In this section, we'll explore `editable`, `toolbars`, and the block `inspector`.
+
+## Editable
+
+At the core of rich text formatting in Gutenberg lies the `Editable` component. It's a wrapper and controller around the TinyMCE library, and can be used by any block to power any rich text editing features. It us used for captions under images or embeds, for the separate fields in a quote, etc.
+
+<ul>
+	<li><strong>Type:</strong> <code>Function</code></li>
```
Comment (aduth) [inline docs/block-components.md]: Technically a function, but more specifically a `Component` class.

File: docs/block-components.md [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+// Within registerBlockType
+edit: function( props ) {
+	var content = props.attributes.content;
+
+	function onChangeContent( newContent ) {
+		props.setAttributes( { content: newContent } );
+	}
+
+	return el(
+		<Editable,
```
Comment (aduth) [inline docs/block-components.md]: Remove leading `<`

File: docs/block-components.md [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+edit( { attributes, className, focus } ) {
+	return (
+		<div className={ className }>
+			<p>{ attributes.content }</p>
+			<Editable
+				tagName="figcaption"
+				placeholder={ __( 'Write caption…' ) }
+				value={ caption }
+				focus={ focus && focus.editable === 'caption' ? focus : undefined }
+				onFocus={ focusCaption }
```
Comment (aduth) [inline docs/block-components.md]: ~straightforward~ 😬

File: docs/block-components.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
+# Tools
+
+To simplify block customization and ensure a consistent experience for users, there are a number of built-in UI patterns to help generate the editor preview. These common tools are offered under the `wp.blocks` global. In this section, we'll explore `editable`, `toolbars`, and the block `inspector`.
+
+## Editable
+
+At the core of rich text formatting in Gutenberg lies the `Editable` component. It's a wrapper and controller around the TinyMCE library, and can be used by any block to power any rich text editing features. It us used for captions under images or embeds, for the separate fields in a quote, etc.
```
Comment (ntwb) [inline docs/block-components.md]: Typo, `It is used for captions` rather than `It us used for captions`

File: docs/block-components.md [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+
+### BlockDescription
+
+A block can display a short text describing what its purpose is. It is recommended to always include one when using the inspector.
+
+# Components
+
+In order to make it easier to create sophisticated blocks quickly, reducing the amount of work you have to do to develop common functionality, several components have been created which can be accessed through the `wp.components` global. This section covers the available UI components.
+
+## Placeholder
```
Comment (gziolo) [inline docs/block-components.md]: Just an idea, should we maybe keep the component related docs in their own README files and only cross-link them in here with a very brief description?

Comment (mtias) [reply docs/block-components.md]: I'd prefer to have them here as this will easier to navigate and refer to.

Comment (youknowriad) [reply docs/block-components.md]: A middle-ground could be to create a separate page for each component using its README

**Comment by gziolo** [comment]
@mtias is it still relevant?

**Comment by mtias** [comment]
@gziolo it'd be nice to generate these from each component's readme. Otherwise it'd be hard to maintain.

**Comment by gziolo** [comment]
Right, it makes a lot of sense what you proposed. I'm closing this one, will open an issue if we don't have one.

**Comment by mtias** [comment]
We can reuse some of the copy here and just start PRs for each component.

**Comment by gziolo** [comment]
@mtias #4952 - feel free to edit description there 👍

**Pull request closed without merging**